### DISCUSSION TITLE
Ability to hook into fields for custom fields

### DISCUSSION THREAD

**Discussion by fvaldes33** [solspace/craft-freeform#22] [Feature Requests]
Is there plan in the roadmap to allow developers to hook into the fields to add our own custom fields?

A use case for us would be to create a select field that pulls in information from a channel or maybe category?

**Comment by kjmartens** [comment]
We may consider this in the future, but we have looked at what it might take to allow for this to happen robustly, and it'd be a fair bit of work on our end. It's on the feature requests list however.

I should also note that features such as loading options from other elements may eventually be included for fields like Checkbox Group, Radio, Select, etc. :)

**Comment by kjmartens** [comment]
I should note that a long time ago, the ability to load field options from other elements was added somewhere along the way in Freeform 2.x. 🙂

**Comment by kjmartens** [comment]
It is possible to create custom field types in Freeform 5.x now. 🙂 
https://docs.solspace.com/craft/freeform/v5/developer/field-types
